### PR TITLE
std/pairing: hinted division in extension fields

### PR DESCRIPTION
**Pull request by yelhousni**
Hinted division in all extension fields and use in bls12377 and bls24315 pairings (G2 and Fpk arithmetics).

```
* BLS12-377
pairing: -263 R1CS contraints
         -613 SCS constraints
G16 verifier: -539 R1CS constraints
         
* BLS24-315
pairing: -809 R1CS constraints
         -2971 SCS constraints
G16 verifier: -1529 R1CS constraints
```

### CODE REVIEW

File: std/algebra/fields_bls12377/e2.go [hunk 1]
```diff
@@ -17,16 +17,26 @@ limitations under the License.
...
+	"math/big"
 )
 
 // E2 element in a quadratic extension
 type E2 struct {
 	A0, A1 frontend.Variable
 }
 
+// SetZero returns a newly allocated element equal to 0
+func (e *E2) SetZero(api frontend.API) *E2 {
```
Comment (gbotrel) [inline std/algebra/fields_bls12377/e2.go:34]: `SetZero` and `SetOne` don't need `api` parameter

File: std/algebra/fields_bls24315/e24.go [hunk 2]
```diff
@@ -93,7 +95,7 @@ func (e *E24) SetOne(api frontend.API) *E24 {
 	e.D0.C0.B1.A1 = 0
 	e.D0.C1.B0.A0 = 0
 	e.D0.C1.B0.A1 = 0
-	e.D0.C1.B1.A0 = 1
+	e.D0.C1.B1.A0 = 0
```
Comment (gbotrel) [inline std/algebra/fields_bls24315/e24.go:98]: was that a mistake in previous version?

Comment (yelhousni) [reply std/algebra/fields_bls24315/e24.go:98]: yes

**Comment by yelhousni** [comment]
count:
```
* BLS12-377
pairing: 11535 R1CS contraints
         54165 SCS constraints
G16 verifier: 19769 R1CS constraints
         
* BLS24-315
pairing: 27608 R1CS constraints
         147158 SCS constraints
G16 verifier: 40666 R1CS constraints
```

**Pull request merged**